app/labor/twitter_bot.rb

```diff
@@ -14,4 +14,44 @@ def client
       config.access_token_secret = @secret
     end
   end
+
+  class << self
+    def fetch(id)
+      retry_on_error(4) { get(id, tweet_mode: "extended") }
+    end
+
+    def get(id, options = {})
+      client.status(id, options)
+    end
+
+    def client
+      @client ||= TwitterBot.new(random_identity).client
+    end
+
+    private
+
+    def random_identity
+      identity = Identity.where(provider: "twitter").last(250).sample
+
+      {
+        token: identity&.token || ApplicationConfig["TWITTER_ACCESS_TOKEN"],
+        secret: identity&.secret || ApplicationConfig["TWITTER_ACCESS_TOKEN_SECRET"]
+      }
+    end
+
+    def retry_on_error(retry_count, &block)
+      yield(block)
+    rescue Twitter::Error => e
+      if retry_count.positive?
+        sleep 0.2
+        retry_count -= 1
+        Rails.logger.error(e)
```

Comment:
I don't know if there's a valid reason to log each error if we're going to retry right after, it's probably enough to log the error at end of the retry count...